services/src/main/java/org/keycloak/validation/DefaultClientValidationProvider.java

```diff
@@ -184,7 +184,7 @@ private void checkUri(FieldMessages field, String url, ValidationContext<ClientM
         }
 
         try {
-            URI uri = new URI(url);
+            URI uri = new URI(url.replace(" ","%20"));
```

Comment:
@DGuhr I just wonder if it wouldn't make more sense to use:

```
url.replaceAll("\\s+","%20")
```

Reply:
definitely makes sense _when we go this way_, but lets answer the bigger question first before moving on i'd say :)

Reply:
(see also the answer to @vmuzikar i gave below)